diff --git a/src/linux/processors.c b/src/linux/processors.c
index 8df06f5..9c7e0a2 100644
--- a/src/linux/processors.c
+++ b/src/linux/processors.c
@@ -12,10 +12,15 @@
 	#include <sched.h>
 #endif
 
+// JEVOIS: some users are not getting CPU_SETSIZE. Maybe __ANDROID__ is defined for them? Here is a fix:
+#ifndef CPU_SETSIZE
+#define CPU_SETSIZE 4
+#endif
+
+
 #include <linux/api.h>
 #include <log.h>
 
-
 #define STRINGIFY(token) #token
 
 #define KERNEL_MAX_FILENAME "/sys/devices/system/cpu/kernel_max"
@@ -78,7 +83,7 @@ inline static bool is_whitespace(char c) {
 		static const uint32_t default_max_processors_count = 32;
 	#endif
 #else
-	static const uint32_t default_max_processors_count = CPU_SETSIZE;
+	static const uint32_t default_max_processors_count = __CPU_SETSIZE;
 #endif
 
 static bool uint32_parser(const char* text_start, const char* text_end, void* context) {
