diff --git a/src/network.c b/src/network.c
index 424057b..225aae6 100644
--- a/src/network.c
+++ b/src/network.c
@@ -182,6 +182,11 @@ network make_network(int n)
 void forward_network(network net)
 {
     int i;
+    char tmp[256]; // JEVOIS
+    
+    jevois_darknet_profiler_start(); // JEVOIS
+
+    
     for(i = 0; i < net.n; ++i){
         net.index = i;
         layer l = net.layers[i];
@@ -193,8 +198,12 @@ void forward_network(network net)
         if(l.truth) {
             net.truth = l.output;
         }
+        snprintf(tmp, 255, "%3d - %s", i, get_layer_string(l.type)); // JEVOIS
+        jevois_darknet_profiler_checkpoint(tmp); // JEVOIS
     }
     calc_network_cost(net);
+
+    jevois_darknet_profiler_stop(); // JEVOIS
 }
 
 void update_network(network net)
