diff --git a/src/network.c b/src/network.c
index aaab799..90d713d 100644
--- a/src/network.c
+++ b/src/network.c
@@ -195,6 +195,9 @@ void forward_network(network *netp)
 #endif
     network net = *netp;
     int i;
+    char tmp[256]; // JEVOIS
+    
+    jevois_darknet_profiler_start(); // JEVOIS
     for(i = 0; i < net.n; ++i){
         net.index = i;
         layer l = net.layers[i];
@@ -206,8 +209,11 @@ void forward_network(network *netp)
         if(l.truth) {
             net.truth = l.output;
         }
+	snprintf(tmp, 255, "%3d - %s", i, get_layer_string(l.type)); // JEVOIS
+	jevois_darknet_profiler_checkpoint(tmp); // JEVOIS
     }
     calc_network_cost(netp);
+    jevois_darknet_profiler_stop(); // JEVOIS
 }
 
 void update_network(network *netp)
@@ -769,6 +775,9 @@ void forward_network_gpu(network *netp)
     }
 
     int i;
+    char tmp[256]; // JEVOIS
+    
+    jevois_darknet_profiler_start(); // JEVOIS
     for(i = 0; i < net.n; ++i){
         net.index = i;
         layer l = net.layers[i];
