diff --git a/src/network.c b/src/network.c
index d56bed5..3763cec 100644
--- a/src/network.c
+++ b/src/network.c
@@ -1,4 +1,5 @@
 #include "darknet.h"
+#include <jevoisbase/Components/ObjectDetection/darknet_profiler.h>
 
 #include <stdio.h>
 #include <time.h>
@@ -256,6 +257,9 @@ void forward_network(network net, network_state state)
 {
     state.workspace = net.workspace;
     int i;
+    char tmp[256]; // JEVOIS
+    
+    jevois_darknet_profiler_start(); // JEVOIS
     for(i = 0; i < net.n; ++i){
         state.index = i;
         layer l = net.layers[i];
@@ -265,8 +269,13 @@ void forward_network(network net, network_state state)
         //double time = get_time_point();
         l.forward(l, state);
         //printf("%d - Predicted in %lf milli-seconds.\n", i, ((double)get_time_point() - time) / 1000);
+
+        snprintf(tmp, 255, "%3d - %s", i, get_layer_string(l.type)); // JEVOIS
+        jevois_darknet_profiler_checkpoint(tmp); // JEVOIS
+
         state.input = l.output;
     }
+    jevois_darknet_profiler_stop(); // JEVOIS
 }
 
 void update_network(network net)
@@ -580,7 +589,7 @@ int resize_network(network *net, int w, int h)
             w = l.out_w;
             h = l.out_h;
         }
-        //if(l.type == AVGPOOL) break;
+        if(l.type == AVGPOOL) break;
     }
 #ifdef GPU
     const int size = get_network_input_size(*net) * net->batch;
