diff --git a/src/convolution-inference.c b/src/convolution-inference.c
index 03b437f..d8d1d27 100644
--- a/src/convolution-inference.c
+++ b/src/convolution-inference.c
@@ -872,6 +872,8 @@ static enum nnp_status compute_gemm_convolution_inference(
 				}
 			}
 			/* Add bias */
+			if (bias)
+			{
 			NNP_OUTPUT_TRANSFORM_START(profile)
 			switch (activation) {
 				case nnp_activation_identity:
@@ -895,6 +897,7 @@ static enum nnp_status compute_gemm_convolution_inference(
 					NNP_UNREACHABLE;
 			}
 			NNP_OUTPUT_TRANSFORM_END(profile)
+			}
 			break;
 		}
 		case nnp_convolution_transform_strategy_precompute:
