diff --git a/tiny_dnn/config.h b/tiny_dnn/config.h
index 685e63a..6dbf327 100644
--- a/tiny_dnn/config.h
+++ b/tiny_dnn/config.h
@@ -13,7 +13,8 @@
 /**
  * define if you want to use intel TBB library
  */
-// #define CNN_USE_TBB
+// JEVOIS: use TBB
+#define CNN_USE_TBB
 
 /**
  * define to enable avx vectorization
@@ -47,6 +48,9 @@
 
 // #define CNN_SINGLE_THREAD
 
+// JEVOIS enable NNPACK:
+#define CNN_USE_NNPACK
+
 /**
  * disable serialization/deserialization function
  * You can uncomment this to speedup compilation & linking time,
@@ -76,7 +80,9 @@
 #ifdef CNN_USE_OMP
 #define CNN_TASK_SIZE 100
 #else
-#define CNN_TASK_SIZE 8
+// JEVOIS assume 48+ CPUs on the host
+//#define CNN_TASK_SIZE 8
+#define CNN_TASK_SIZE 50
 #endif
 
 namespace tiny_dnn {
@@ -85,11 +91,14 @@ namespace tiny_dnn {
  * calculation data type
  * you can change it to float, or user defined class (fixed point,etc)
  **/
-#ifdef CNN_USE_DOUBLE
-typedef double float_t;
-#else
+//JEVOIS: use float instead of default double:
 typedef float float_t;
-#endif
+
+//#ifdef CNN_USE_DOUBLE
+//typedef double float_t;
+//#else
+//typedef float float_t;
+//#endif
 
 /**
  * size of layer, model, data etc.
